jvm/src/main/java/com/microsoft/bond/protocol/FastBinaryReader.java

```diff
@@ -95,70 +92,219 @@ public short readInt16() throws IOException {
+            case BondDataType.Values.BT_FLOAT:
+                this.reader.skipBytes(4);
+                break;
+
+            // 8-byte
+            case BondDataType.Values.BT_UINT64:
+            case BondDataType.Values.BT_INT64:
+            case BondDataType.Values.BT_DOUBLE:
+                this.reader.skipBytes(8);
+                break;
+
+            // UTF-8 string (1-byte Unicode code units)
+            case BondDataType.Values.BT_STRING:
+                this.reader.skipBytes(this.reader.readVarUInt32());
+                break;
+
+            // UTF-16 string (2-byte Unicode code units)
+            case BondDataType.Values.BT_WSTRING:
+                this.reader.skipBytes(this.reader.readVarUInt32() * 2L);
+                break;
+
+            // List/Set collections (recursive)
+            case BondDataType.Values.BT_LIST:
+            case BondDataType.Values.BT_SET:
+                this.skipListContainer();
+                break;
+
+            // Map collections (recursive)
+            case BondDataType.Values.BT_MAP:
+                this.skipMapContainer();
+                break;
+
+            // Nested struct (recursive)
+            case BondDataType.Values.BT_STRUCT:
+                this.skipStruct();
+                break;
+
+            // Invalid type
+            default:
+                throw new InvalidBondDataException("Invalid Bond data type: " + type);
+        }
+    }
+
+    private void skipListContainer() throws IOException {
+        final BondDataType elementType = this.readType();
+        int count = this.reader.readVarUInt32();
+
+        // Process fixed-width data types separately to avoid looping over all elements
+        int elementTypeFixedWidth = getFixedTypeWidth(elementType);
+        if (elementTypeFixedWidth > 0) {
+            this.reader.skipBytes((long) count * elementTypeFixedWidth);
+        } else {
+            while (--count >= 0) {
+                this.skip(elementType);
+            }
+        }
+    }
+
+    private void skipMapContainer() throws IOException {
+        final BondDataType keyType = this.readType();
+        final BondDataType elementType = this.readType();
+        int count = this.reader.readVarUInt32();
+
+        // Process fixed-width data types separately to avoid looping over all elements
+        int keyTypeFixedWidth = getFixedTypeWidth(elementType);
+        int elementTypeFixedWidth = getFixedTypeWidth(elementType);
+        if (keyTypeFixedWidth > 0 && elementTypeFixedWidth > 0) {
+            this.reader.skipBytes((long) count * (keyTypeFixedWidth + elementTypeFixedWidth));
+        } else {
+            while (--count >= 0) {
+                this.skip(keyType);
+                this.skip(elementType);
+            }
+        }
+    }
+
+    private void skipStruct() throws IOException {
+        while (true) {
+            BondDataType fieldType = this.readType();
+
+            if (fieldType.value == BondDataType.BT_STOP_BASE.value)
```

Comment:
Yes.

---
In reply to: [119243725](https://github.com/Microsoft/bond/pull/465#discussion_r119243725) [](ancestors = 119243725)